perftest/COOLPerfTest.cpp

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include <sysexits.h>
+#include <filesystem>
+#include <memory>
+
+#include <Poco/Util/Application.h>
+#include <Poco/URI.h>
+
+#include <net/Ssl.hpp>
+#if ENABLE_SSL
+#  include <SslSocket.hpp>
+#endif
+#include <Log.hpp>
+
+#include <perftest/PerfTest.hpp>
+#include <perftest/SamplePerfTest.hpp>
```

Comment:
"perftest/COOLPerfTest.cpp currently includes perftest/SamplePerfTest.hpp" still needs to be resolved, I guess.

Reply:
Got dlopen working, like in the unit tests. This was a pain to get working, but before I got it I also tried to extend the header includes and it got really messy.

I had come up with these requirements earlier, and dlopen meets them much better than header includes would have.
- Ability to run one performance test
- Ability to run all performance tests
- Adding a test should only require touching one other file, and only a few lines there
- Adding a test should require little or no boilerplate
- Test files should be able to go in nested directories